code/datums/diseases/advance/symptoms/tumors.dm

```diff
@@ -0,0 +1,102 @@
+//benign, premalignant, malignant tumors
+
+/datum/symptom/tumor
+	name = "Benign tumors"
+	desc = "The virus causes benign growths all over your body."
+	stealth = 0
+	resistance = 4
+	stage_speed = -4
+	transmittable = -4
+	level = 3
+	severity = 2
+	symptom_delay_min = 5
+	symptom_delay_max = 35
+	threshold_descs = list(
+		"Transmission 7" = "Gives visible growths on the host's body, eventually preventing some clothes from being worn.",
+		"Stealth 4" = "Regenerates limbs that are incredibly fragile.",
+		"Resistance 8" = "Heals brute and burn damage in exchange for toxin damage."
+	)
+	var/regeneration = FALSE
+	var/helpful = FALSE
+	var/tumor_chance = 0.5
+	var/obj/item/organ/tumor/tumortype = /obj/item/organ/tumor
+	var/datum/disease/advance/ownerdisease //what disease it comes from
+
+/datum/symptom/tumor/Start(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return
+	ownerdisease = A;
+	if(A.totalTransmittable() >= 7) //visible growths
+		if(ishuman(A.affected_mob))
+			A.affected_mob.visible_tumors = TRUE
+	if(A.totalStealth() >= 4) //regeneration of limbs
+		regeneration = TRUE
+	if(A.totalResistance() >= 8) //helpful healing instead of just toxin
+		helpful = TRUE
+
+/datum/symptom/tumor/Activate(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/carbon/human/M = A.affected_mob
+	if(!M) return
+
+	if(M.visible_tumors)
+		//clothes wearing
+		if(A.stage > 2)
+			var/datum/species/S = M.dna?.species
+			if(S)
+				S.add_no_equip_slot(M, SLOT_WEAR_MASK)
+				S.add_no_equip_slot(M, SLOT_HEAD)
+
+		if(A.stage == 5)
+			var/datum/species/S = M.dna?.species
+			if(S)
+				S.add_no_equip_slot(M, SLOT_WEAR_SUIT)
+
+	//spreading
+	if(prob(tumor_chance)) //2% chance to make a new tumor somewhere
```

Comment:
please don't have this repeated in two places.

Reply:
where do you suggest it goes

Reply:
i don't care, the ugly way if you cant find a way to properly OOP (which i don't think we can) would just be a global proc or something in this file